### PR TITLE
[web] ProxyApp and MainView with ES6

### PR DESCRIPTION
**Pull request by gzzhanghao**
There are lots of things we need to deal with, but let's start with components first :)

### CODE REVIEW

File: web/src/js/components/flowview/index.js [hunk 1]
```diff
@@ -89,7 +89,6 @@ var FlowView = React.createClass({
             } else {
                 active = tabs[0];
             }
-            this.selectTab(active);
```
Comment (mhils) [inline web/src/js/components/flowview/index.js]: Was this change intentional?

**Comment by mhils** [comment]
Looks great, happy to see you pushing forward!

Any explanation why your compiled JS and CSS files are so much smaller?

**Comment by gzzhanghao** [comment]
@mhils I used `gulp prod` to minimize script files, we should do this everytime we push a commit, but I didn't realize that before :P

File: web/src/js/components/flowview/index.js (hunk 1 above)
Comment (gzzhanghao) [inline web/src/js/components/flowview/index.js]: I'd like to keep `render` pure.
Also, it leads to an issue when `selectedFlow` is not sync with `routerParams.flowId`

Comment (mhils) [inline web/src/js/components/flowview/index.js]: Ah yes - saw the purity issue as well, but didn't  get back to it. Good find.
